plugins/storage/volume/scaleio/src/main/java/org/apache/cloudstack/storage/datastore/provider/ScaleIOHostListener.java

```diff
@@ -171,11 +171,7 @@ private ModifyStoragePoolAnswer sendModifyStoragePoolCommand(ModifyStoragePoolCo
 
     @Override
     public boolean hostDisconnected(long hostId, long poolId) {
-        StoragePoolHostVO storagePoolHost = _storagePoolHostDao.findByPoolHost(poolId, hostId);
-        if (storagePoolHost != null) {
-            _storagePoolHostDao.deleteStoragePoolHostDetails(hostId, poolId);
-        }
-
```

Comment:
@harikrishna-patnala I think, better to remove the record on host disconnect as it gets updated (with any SDC Id changes) on host re-connect. Any issue, if it is deleted from _storage_pool_host_ref_? 

Other storage providers also remove it on disconnect, check here:
https://github.com/apache/cloudstack/blob/f3ad9e6743a01bc32c54dd7549330723dbf895f6/plugins/storage/volume/datera/src/main/java/org/apache/cloudstack/storage/datastore/provider/DateraHostListener.java#L115

https://github.com/apache/cloudstack/blob/f3ad9e6743a01bc32c54dd7549330723dbf895f6/plugins/storage/volume/solidfire/src/main/java/org/apache/cloudstack/storage/datastore/provider/SolidFireSharedHostListener.java#L120

Reply:
yes @sureshanaparti there is some race condition issue, entry in storage_pool_host_ref is getting deleted and the new entry is not getting added during host connect. Since we are dealing with SDC ID updation and creating a new entry if does not exist during host connect, I've removed the deleting part in host disconnect.

If the entry of SDC ID is not there in storage_pool_host_ref all Powerflex operations are affected.

Reply:
> yes @sureshanaparti there is some race condition issue, entry in storage_pool_host_ref is getting deleted and the new entry is not getting added during host connect. Since we are dealing with SDC ID updation and creating a new entry if does not exist during host connect, I've removed the deleting part in host disconnect.
> 
> If the entry of SDC ID is not there in storage_pool_host_ref all Powerflex operations are affected.

@harikrishna-patnala What is that race condition issue? It's fine if dealing with SDC ID updation on host connect. What if the host is removed, and not connected again. Check if these entries of the host are removed from _storage_pool_host_ref_  or not. If host is connected, ensure that SDC ID is properly added/updated for that host in _storage_pool_host_ref_, then no Powerflex operations are affected.

Reply:
I think race condition happening in multiple management server setup. Yes all entries are getting deleting when hist is removed, so upon adding it again new entries will be made. Even if SDC ID is changed which is unlikely event, even then it will be updated during host connect. @sureshanaparti